contracts/handler/CompatibilityFallbackHandler.sol

```diff
@@ -62,7 +62,9 @@ contract CompatibilityFallbackHandler is TokenCallbackHandler, ISignatureValidat
         if (_signature.length == 0) {
             require(safe.signedMessages(messageHash) != 0, "Hash not approved");
         } else {
-            safe.checkSignatures(messageHash, _signature);
+            // We explicitly do not allow caller approved sigantures for EIP-1271. This is done by setting the executor
```

Comment:
nit + typo:
```suggestion
            // We explicitly do not allow caller approved signatures for EIP-1271 to prevent unexpected behaviour. This is done by setting the executor
```

Reply:
maybe also worth mentioning that we couldn't find a valid use case for it